cmd/skopeo/sync.go

```diff
@@ -0,0 +1,509 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/image/copy"
+	"github.com/containers/image/docker"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/manifest"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/transports/alltransports"
+	"github.com/containers/image/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+type repoDescriptor struct {
+	RootReference  types.ImageReference
+	TagsReferences []types.ImageReference
+	Context        *types.SystemContext
+}
+
+type registrySyncCfg struct {
+	Images      map[string][]string
+	Credentials types.DockerAuthConfig
+	TLSVerify   bool   `yaml:"tls-verify"`
+	CertDir     string `yaml:"cert-dir"`
+}
+
+type sourceCfg map[string]registrySyncCfg
+
+func newSourceConfig(yamlFile string) (cfg sourceCfg, err error) {
+
+	err = yamlUnmarshal(yamlFile, &cfg)
+	return
+}
+
+func isValidTransport(transport types.ImageTransport) (bool, error) {
+	dockerTransport := transports.Get("docker")
+	if dockerTransport == nil {
+		return false, fmt.Errorf("Cannot find 'docker' transport type")
+	}
+
+	dirTransport := transports.Get("dir")
```

Comment:
could we add to `validTransports` only the ones that are present?  i.e. if dirTransport is missing we just skip it and not add it to `validTransports` as anyway the `docker` -> `docker` case doesn't need it.